types/src/account_config/events/new_block.rs

```diff
@@ -76,3 +86,41 @@ impl MoveStructType for NewBlockEvent {
     const MODULE_NAME: &'static IdentStr = ident_str!("block");
     const STRUCT_NAME: &'static IdentStr = ident_str!("NewBlockEvent");
 }
+
+pub fn new_block_event_key() -> EventKey {
+    EventKey::new(2, CORE_CODE_ADDRESS)
+}
+
+/// The path to the new block event handle under a Block::BlockMetadata resource.
+pub static NEW_BLOCK_EVENT_PATH: Lazy<Vec<u8>> = Lazy::new(|| {
+    let mut path = BlockResource::resource_path();
+    // it can be anything as long as it's referenced in AccountState::get_event_handle_by_query_path
+    path.extend_from_slice(b"/new_block_event/");
+    path
+});
+
+/// Should be kept in-sync with BlockMetadata move struct in block.move.
+/// TODO: Should be renamed to BlockMetadata to match.
```

Comment:
There's a TODO from earlier to rename BlockMetadata in block.move to BlockResource. So which one are you using for both lol - BlockResource or BlockMetadata?